### ISSUE TITLE
CC ToEnum doesn't work properly

### ISSUE THREAD

**Issue by watchforstock** [ninjasphere/go-openzwave#7]
Hi,

Firstly thanks for your work on this library - I wouldn't be able to use openzwave from Go without it!

I noticed that when converting numbers to the CC Enum, I was always getting UNKNOWN. I think the problem is in the init() function (https://github.com/ninjasphere/go-openzwave/blob/master/scripts/GenerateCC.sh#L48)

All of the map values are set to the same point (&e) and the last value that is set to is UNKNOWN. As a workaround, I changed the init function like this:

```
func init() {
     for _, e := range enums {
        ee = Enum{e.Code, e.Name}
        fromName[e.Name] = &ee
        fromCode[e.Code] = &ee
     }
}
```

The above works, but doesn't feel like the most satisfactory answer. Any thoughts on the best way to fix this?

Thanks

**Comment by Troy-M** [comment]
Can you show the file CC/CC.go? That should help debug where this issue is coming from.

**Comment by watchforstock** [comment]
Please see attached (renamed as CC.txt for github), noting that this includes my changes (to the init() function as detailed above.
[CC.txt](https://github.com/ninjasphere/go-openzwave/files/512375/CC.txt)

**Comment by Troy-M** [comment]
Can you confirm switching the init function to

``` golang
func init() {
     for i, _ := range enums {
        e := enums[i]
        fromName[e.Name] = &e
        fromCode[e.Code] = &e
     }
}
```

Fixes the problem? See [this](http://stackoverflow.com/a/29498133) SO for the reasoning.

**Comment by jonseymour** [comment]
Sorry about that @watchforstock, thank @Troy-M.

Should be fixed by 24fe2ea
